ProjectTemplates/templates/Worker-CSharp/GreeterService.cs

```diff
@@ -13,26 +23,87 @@ namespace Company.Application1
     public class GreeterService : Greeter.GreeterBase
     {
         private readonly ILogger<GreeterService> _logger;
-        public GreeterService(ILogger<GreeterService> logger)
+
+#if (!NoAuth)
+        // The web API will only accept tokens 1) for users, and 2) having the "api-scope" scope for this API
+        static readonly string[] scopeRequiredByApi = new string[] { "api-scope" };
+#endif
+
+#if (GenerateApi)
+        private readonly IDownstreamWebApi _downstreamWebApi;
+
+        public GreeterService(ILogger<GreeterService> logger,
+                        IDownstreamWebApi downstreamWebApi)
         {
             _logger = logger;
+            _downstreamWebApi = downstreamWebApi;
         }
 
-        static string[] scopeRequiredByAPI = new string[] { "access_as_user" };
+        [Authorize]
+        public override Task<HelloReply> SayHello(HelloRequest request, ServerCallContext context)
+        {
+            var httpContext = context.GetHttpContext();
+            httpContext.VerifyUserHasAnyAcceptedScope(scopeRequiredByApi);
+            using var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                var apiResult = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                // Do something with apiResult
+            }
+            else
+            {
+                var error = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                throw new HttpRequestException($"Invalid status code in the HttpResponseMessage: {response.StatusCode}: {error}");
+            }
+
+            return Task.FromResult(new HelloReply
+            {
+                Message = "Hello " + request.Name
+            });
+        }
+
+# elseif (GenerateGraph)
```

Comment:
There is an extra space between `#` and `elseif`.

Reply:
yes, nice catch. thanks.